lib/ui/painting/image_decoder.cc

```diff
@@ -57,30 +62,91 @@ static SkISize GetResizedDimensions(SkISize current_size,
 }
 
 static sk_sp<SkImage> ResizeRasterImage(sk_sp<SkImage> image,
+                                        std::unique_ptr<SkCodec> codec,
                                         std::optional<uint32_t> target_width,
                                         std::optional<uint32_t> target_height,
                                         const fml::tracing::TraceFlow& flow) {
   FML_DCHECK(!image->isTextureBacked());
 
-  const auto resized_dimensions =
+  TRACE_EVENT0("flutter", __FUNCTION__);
+  flow.Step(__FUNCTION__);
+
+  if (!target_width && !target_height) {
+    // The resized dimesions are the same as the intrinsic dimensions of the
+    // image. There is nothing to do.
+    return image->makeRasterImage();
+  }
+
+  auto resized_dimensions =
       GetResizedDimensions(image->dimensions(), target_width, target_height);
 
   if (resized_dimensions.isEmpty()) {
     FML_LOG(ERROR) << "Could not resize to empty dimensions.";
     return nullptr;
   }
 
-  if (resized_dimensions == image->dimensions()) {
-    // The resized dimesions are the same as the intrinsic dimensions of the
-    // image. There is nothing to do.
-    return image;
+  // This is where the scaling happens.
+  if (resized_dimensions.width() > image->dimensions().width() ||
+      resized_dimensions.height() > image->dimensions().height()) {
+    FML_LOG(WARNING)
+        << "Image is being upsized. Using cache(Height|Width) correctly?";
```

Comment:
We could make this warning more actionable by indicating what the actual height/width of the image is and the requested height width.

Reply:
Done.